OpenAdaptAI/OpenAdapt
# Add posthog analytics to tray actions

**What kind of change does this PR introduce?**
This PR introduces posthog analytics to server actions



**Summary**
This PR adds a couple of things
- Adds a unique identifier for users (set by running migrations) to the config, so its easier to group events from different users
- Adds psothog events to server side functions and tray buttons


#715 

**Checklist**

**How can your code be run and tested?**
To test locally, search for `posthog.disabled` in `openadapt/utils.py` and comment out that code block. Also in `openadapt/app/dashboard/run.py`, set `NEXT_PUBLIC_MODE` to "production" unconditionally, just to mimic what the app would behave like when run from the build.

Then run `python -m openadapt.entrypoint` (which is the usual command to run the app) and the new migration should create a unique user id in your `config.json` file. Play around the tray and the dashboard, and you should be able to see posthog events coming in tied to your user id (https://us.posthog.com/project/68185/persons)
